torchdynamo/guards.py

```diff
@@ -273,34 +336,83 @@ def PYMODULE_MATCH(self, guard: Guard):
     def LIST_LENGTH(self, guard):
         ref = self.arg_ref(guard)
         value = self.get(guard.name)
-        self.code.append(f"___check_type_id({ref}, {self.id_ref(type(value))})")
-        self.code.append(f"len({ref}) == {len(value)}")
+        t = type(value)
+
+        code_type_check = f"___check_type_id({ref}, {self.id_ref(t)})"
+        self.code.append(code_type_check)
+        code_len_check = f"len({ref}) == {len(value)}"
+        self.code.append(code_len_check)
+
+        guard.set_export_info(
+            "LIST_LENGTH", t, f"{code_type_check} and {code_len_check}", id(value)
```

Comment:
This pattern of calling `set_export_info` is a bit ugly as now there is a lot of redundancy in guard setup that could be easy for things to get out of sync.

I wonder if we could have a helper method that both updates `self.code` and the export info.  We might be able to automate things like the guard name.  

It could be more useful to provide pointers to objects rather than id(value).